tools/cli/commands/changelog.js

```diff
@@ -512,25 +518,31 @@ async function gitAdd( argv ) {
 }
 
 /**
- * Gets list of currently modified files.
+ * Checks if changelog files are required.
  *
- * @returns {Array} modifiedProjects - projects that need a changelog.
+ * @returns {Array} matchedProjects - projects that need a changelog.
  */
-async function changedProjects() {
-	const re = /^projects\/([^/]+\/[^/]+)\//; // regex matches project file path, ie 'project/packages/connection/..'
-	const modifiedProjects = new Set();
-	const git = simpleGit();
-	const gitStatus = await git.status();
-	for ( const file of gitStatus.files ) {
-		const match = file.path.match( re );
-		if ( match ) {
-			modifiedProjects.add( match[ 1 ] );
-		}
+async function checkChangelogFiles() {
+	console.log( chalk.green( 'Checking if changelog files are needed. Just a sec...' ) );
+
+	// Bail if we're pushing to a release branch, like boost/branch-1.3.0
+	let currentBranch = child_process.spawnSync( 'git', [ 'branch', '--show-current' ] );
+	currentBranch = currentBranch.stdout.toString().trim();
+	const branchReg = /.*\/branch-(\d+).(\d+)(.(\d+))?/; // match example: jetpack/branch-1.2.3
```

Comment:
The `.*` at the beginning isn't needed, since the pattern isn't anchored. But you do need to escape the `.` that are intended to be literal.
```suggestion
	const branchReg = /\/branch-(\d+)\.(\d+)(\.(\d+))?/; // match example: jetpack/branch-1.2.3
```
I also note this will match something like "jetpack/branch-1.2-beta". I'm guessing that's intended, but if not you should anchor it with a `$` at the end of the pattern.

Reply:
ping?

Reply:
Whoops, forgot the first part there. As for the branch matching, I think that would be expected unless anyone thinks otherwise.

Reply:
If anything I'd go in the opposite direction. Most of the rest of our tooling doesn't care if it's "/branch-foobar" instead of having a version number at all.